shotover-proxy/src/transforms/cassandra/peers_rewrite.rs

```diff
@@ -67,15 +78,60 @@ fn rewrite_port(message: &mut Message, new_port: u32) {
         if let CassandraOperation::Result(CassandraResult::Rows { value, metadata }) =
             &mut frame.operation
         {
-            let port_column_index = metadata
-                .col_specs
+            let columns = ["native_port", "rpc_port", "broadcast_port", "peer_port"];
+
+            let port_column_indices = columns
                 .iter()
-                .position(|col| col.name.as_str() == "native_port");
+                .map(|col| {
+                    metadata
+                        .col_specs
+                        .iter()
+                        .position(|wanted_col| &wanted_col.name.as_str() == col)
+                })
+                .flatten()
+                .collect::<Vec<_>>();
```

Comment:
we dont need to collect here

Reply:
Did you mean use `filter_map` instead?

Reply:
I mean like 
![image](https://user-images.githubusercontent.com/5120858/157568387-32150c73-5918-4536-a340-ea9cf0b38ece.png)